### PR TITLE
*: replace global mutex random with lock-free FastRand function.

### PR DESCRIPTION
**Pull request by coocood**
### What problem does this PR solve? 
Improve the perfromance.

### What is changed and how it works?
replace global mutex random with lock-free FastRand function.

```
   BenchmarkFastRand-12    	1000000000	         0.435 ns/op
   BenchmarkGlobalRand-12    	18839404	     61.8 ns/op
```

### Check List 

Tests 

 - Unit test

### CODE REVIEW

**Comment by coocood** [comment]
@bobotu @tiancaiamao PTAL

**Comment by bobotu** [comment]
LGTM

File: util/random_test.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+}
+
+func (s *testMiscSuite) TestRand(c *C) {
+	x := FastRand32N(1024)
+	c.Assert(x < 1024, IsTrue)
+	y := FastRand64N(1 << 63)
+	c.Assert(y < 1<<63, IsTrue)
+
+	_ = RandomBuf(20)
+}
```
Comment (tiancaiamao) [inline util/random_test.go]: Test data uniform distributed?

```
var arr[1024]bool
for i:=0; i<1024; i++ {
    idx := FastRand32N(1024)
    arr[idx] = true
}
sum := 0
for i:=0; i<1024; i++ {
    if arr[i] == false {
        sum++
    }
}
c.Assert(sum < 24)
```

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**